### PR TITLE
Consuming the TinEye Reverse Image Search API in Node.js

### LINKED ISSUE #1882: How to perform a reverse image search in Node.js using the TinEye Reverse Image Search API
### Proposed title of the article
Using the TinEye Reverse Image Search API in Node.js

### Introduction paragraph (2-3 paragraphs):
At one point or another, we get a need of searching online for an image. Perhaps we need one with better quality, or we need to see the origin of the image or any other reason. Reverse image searching now comes in and here we are going to delve into the API of one of the commonly used tools, the TinEye reverse image search. We will look at how to feed images and get the results using Node.js.

### Key takeaways: 
In the end, the reader will learn about:

- Reverse Image Searching
- The TinEye Reverse Image Search
- Consuming the TinEye Reverse Image Search API
- Troubleshooting common errors that occur during the process

### References: 
[TinEye API documentation](https://services.tineye.com/developers/tineyeapi/what_is_tineyeapi)

### PR DESCRIPTION
**Pull request by munubi254**
### Article checklist
**NOTE: (Please ensure that you have only one open issue + linked pull request at a time. This will ensure that we complete the article in a timely manner from inception to publishing.)**
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) -  [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker? We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

This fixes #1882

### CODE REVIEW

**Comment by munubi254** [comment]
@paulodhiambo Added author page

File: articles/node-tineye-api/index.md [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+1. TinEye
+2. Google Images
+3. Pixsy
+
+In this article, we are going to talk about the TinEye Reverse Image Search and how to consume its API in Node.js. It is a RESTFUL API.
+
+#### Prerequisites
+
+1. A basic understanding of Node.js and JavaScript concepts
+2. Node.js(NPM) installed in your system
```
Comment (paulodhiambo) [inline articles/node-tineye-api/index.md:32]: Add links to Node.js, JavaScript  documentations.

**Review by paulodhiambo — CHANGES_REQUESTED** [review]
@munubi254 review the change requested and ping me when done.

**Comment by munubi254** [comment]
@paulodhiambo Done. Please check.

**Comment by paulodhiambo** [comment]
@munubi254 amazing article. @hectorkambow you can give it a final review.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by hectorkambow** [comment]
great work @munubi254  🚀  
very nice tutorial - seems like a helpful addition to the EngEd library.
cc @paulodhiambo

**Pull request merged**